src/OpenTelemetry.Instrumentation.Hangfire/Implementation/HangfireInstrumentationJobFilterAttribute.cs

```diff
@@ -65,5 +79,38 @@ public void OnPerformed(PerformedContext performedContext)
                 activity.Dispose();
             }
         }
+
+        public void OnCreating(CreatingContext creatingContext)
+        {
+            // Short-circuit if nobody is listening
+            if (!HangfireInstrumentation.ActivitySource.HasListeners() || Activity.Current?.Id is null)
```

Comment:
why checking if Activity.Current is null here? The same check seems repeated few lines below as well.

Reply:
Good catch, fixed it